### PR TITLE
V0.3.5

### PR DESCRIPTION
**Pull request by SC426**
ruff 0.3.5

**Destination channel:** defaults

### Links

- [PKG-4208](https://anaconda.atlassian.net/browse/PKG-4208)
- [Upstream repository](https://github.com/astral-sh/ruff/tree/v0.2.2)
- [Upstream changelog/diff](https://github.com/astral-sh/ruff/blob/v0.2.2/CHANGELOG.md)

### Explanation of changes:

- Add new submodule
- Linter fixes

[PKG-4208]: https://anaconda.atlassian.net/browse/PKG-4208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
-    - make
+    - make  # [unix]
     - maturin
     - cargo-bundle-licenses
   host:
     - python
     - pip
     - maturin
     - toml
+    - wheel
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```
I think wheel isn't required as a build backend is `maturin`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - python
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ compiler('rust') }}
-    - make
+    - make  # [unix]
     - maturin
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - maturin >=1.0,<2.0
```

**Review by cbouss — COMMENTED** [review]
You should be able to remove make and the c and cxx compilers from the build sedction too. Because of https://github.com/AnacondaRecipes/rust-activation-feedstock/blob/master/recipe/meta.yaml#L21-L23

**Comment by SC426** [comment]
Screenshot 2024-04-16 at 10 16 36 AM

I think we may need to keep `make`

**Review by skupr-anaconda — APPROVED** [review]

**Review by JamesRobertsonGames — APPROVED** [review]
Looks good to me! :+1:

**Review by cbouss — APPROVED** [review]

**Pull request merged**